src/Packetery/Module/Order/LabelPrint.php

```diff
@@ -354,23 +345,27 @@ private function requestPacketaLabels( int $offset, array $packetIds ): Response
 	}
 
 	/**
-	 * @param int     $offset
-	 * @param array[] $packetIdsWithCourierNumbers
+	 * @param int                                                        $offset
+	 * @param LabelPrintPacketData                                       $labelPrintPacketData
+	 * @param array<int, array{packetId: string, courierNumber: string}> $packetIdsWithCourierNumbers
+	 *
+	 * @return Response\PacketsCourierLabelsPdf
 	 */
-	private function requestCarrierLabels( int $offset, array $packetIdsWithCourierNumbers ): Response\PacketsCourierLabelsPdf {
+	private function requestCarrierLabels( int $offset, LabelPrintPacketData $labelPrintPacketData, array $packetIdsWithCourierNumbers ): Response\PacketsCourierLabelsPdf {
 		$request  = new Request\PacketsCourierLabelsPdf( array_values( $packetIdsWithCourierNumbers ), $this->labelPrintParametersService->getLabelFormat(), $offset );
 		$response = $this->soapApiClient->packetsCarrierLabelsPdf( $request );
 
 		foreach ( $packetIdsWithCourierNumbers as $orderId => $pairItem ) {
 			$record          = new Log\Record();
 			$record->action  = Log\Record::ACTION_CARRIER_LABEL_PRINT;
 			$record->orderId = $orderId;
-			$order           = $this->orderRepository->getByIdWithValidCarrier( $orderId );
+			$order           = $labelPrintPacketData->getOrderByOrderId( $orderId );
+			if ( $order === null ) {
```

Comment:
Tady říkáme, že může dojít k situaci, kdy nezalogujeme výsledek tisku štítku k nějaké objednávce, protože nemáme objednávku.   To je situace, kterou jsme chtěli tímto refactoringem odstranit.  A dle mého názoru k tomu ani nemůže dojít, pokud getOrderByOrderId() nebude vracet null (viz komentář https://github.com/Zasilkovna/WooCommerce/pull/906/changes#r2630799050)